libs/libsysnds/source/arm9/fatfs/diskio.c

```diff
@@ -299,6 +316,7 @@ DRESULT disk_read(BYTE pdrv, BYTE *buff, LBA_t sector, UINT count)
 // count:  Number of sectors to write
 DRESULT disk_write(BYTE pdrv, const BYTE *buff, LBA_t sector, UINT count)
 {
+    uint8_t align_buffer[FF_MAX_SS];

```

Comment:
Allocating this in the stack seems like a bad idea, isn't that in DTCM? That wouldn't work for DSi writes because they are done with a DMA, if I remember correctly.

Reply:
... stacks are in DTCM? That'd be news to me... but sure, I can just `malloc()` it and bail on failure...